backend/media/src/main/java/co/airy/core/media/services/MediaUpload.java

```diff
@@ -14,22 +21,30 @@
 import java.net.MalformedURLException;
 import java.net.URL;
 import java.net.URLConnection;
+import java.util.List;
 
 @Service
 @EnableRetry
-public class MediaUpload {
+public class MediaUpload implements HealthIndicator {
+    private static final Logger log = AiryLoggerFactory.getLogger(MediaUpload.class);
     private final AmazonS3 amazonS3Client;
     private final String bucket;
     private final String path;
     private final URL host;
 
+    @Getter
+    private boolean connectionStatus;
+
     public MediaUpload(AmazonS3 amazonS3Client,
                        @Value("${s3.bucket}") String bucket,
                        @Value("${s3.path}") String path) throws MalformedURLException {
         this.amazonS3Client = amazonS3Client;
         this.bucket = bucket;
         this.path = appendSlash(path);
         this.host = new URL(String.format("https://%s.s3.amazonaws.com/", bucket));
+
+        // Check if the connection is healthy
```

Comment:
I'd try to limit the use of comments to when the logic is not deducible from the code

Reply:
I think even if it is a bit redundant. That make it obvious and easier to the eye to see. In this case it easy to know that the check it is only done during the construction face.

Reply:
Alright, I don't have a strong opinion so yours that you did the feature matter more :)